### ISSUE TITLE
Using commands.spawn in a non-startup system causes erratic behavior

### ISSUE THREAD

**Issue by Laremere** [bevyengine/bevy#440]
I have a sample which creates two circles of spheres: one in system startup, and one in a normal (?) system.
For each spawn call, I am creating a new material so you can see the order the spheres were constructed in.
The startup system successfully creates a nice range of colors for the spheres:

![image](https://user-images.githubusercontent.com/202599/92299767-46842c00-ef0a-11ea-9986-b06c52cc97ca.png)

However once the spheres created by spawn_spheres start spawning, things go wonky and the colors change a bunch:
![image](https://user-images.githubusercontent.com/202599/92299845-f194e580-ef0a-11ea-8497-5b0123a9aed1.png)

The colors mostly just swap around, but occasionally are duplicated.

My expectation is that spawning both in setup and spawn_spheres should behave the same.

I also was experiencing a panic with the code before I simplified (with a different component besides the material), though it seems to be triggered under the same condition, so not creating an issue for that as well.

```rust
use bevy::prelude::*;

fn main() {
    App::build()
        .add_resource(R(0.0))
        .add_default_plugins()
        .add_startup_system(setup.system())

        .add_system(spawn_spheres.system())

        .run();
}

fn setup(
    mut commands: Commands,
    mut meshes: ResMut<Assets<Mesh>>,
    mut materials: ResMut<Assets<StandardMaterial>>,
) {
    commands
        .spawn(LightComponents {
            translation: Translation::new(4.0, 8.0, 4.0),
            ..Default::default()
        })
        .spawn(Camera3dComponents {
            	transform: Transform::new_sync_disabled(Mat4::face_toward(
                	Vec3::new(-3.0, 5.0, 8.0),
                	Vec3::new(0.0, 0.0, 0.0),
                	Vec3::new(0.0, 1.0, 0.0),
            )),
            ..Default::default()
        })
        ;


 	let mut r = 0.0;
    while r < std::f32::consts::PI * 2.0 {
		let value = r / (std::f32::consts::PI * 2.0);
		commands.spawn(PbrComponents {
            mesh: meshes.add(Mesh::from(shape::Icosphere {
                subdivisions: 4,
                radius: 0.5,
            })),
            material: materials.add(Color::rgb(value, value, value).into()),
            translation: Translation::new(r.cos() * 4.0,0.0, r.sin() * 4.0),
            ..Default::default()
        });
		r += std::f32::consts::PI / 10.0;
	}
}

struct R(f32);

fn spawn_spheres(
	mut commands: Commands,
    mut meshes: ResMut<Assets<Mesh>>,
    mut materials: ResMut<Assets<StandardMaterial>>,
    mut r: ResMut<R>,
    time: Res<Time>,
) {
	if r.0 * 5.0 + 5.0 < time.seconds_since_startup as f32 {
		if r.0 < std::f32::consts::PI * 2.0 {
			let value = r.0 / (std::f32::consts::PI * 2.0);
			commands.spawn(PbrComponents {
	            mesh: meshes.add(Mesh::from(shape::Icosphere {
	                subdivisions: 4,
	                radius: 0.5,
	            })),
	            material: materials.add(Color::rgb(value, value, value).into()),
	            translation: Translation::new(r.0.cos() * 4.0,2.0, r.0.sin() * 4.0),
	            ..Default::default()
	        });
		}
		r.0 += std::f32::consts::PI / 10.0;
	}
}
```

**Comment by memoryruins** [comment]
Thanks for reporting! I think this was fixed recently (similar to #328). With your code, I am able to recreate the issue with version `0.1.3`, but with master branch 5f1fef333cfddde1d4dbbc223338f7ea0edc0875, it now behaves correctly.

![image](https://user-images.githubusercontent.com/6868531/92300894-b6a3a980-ef2c-11ea-8ac8-121c978a7d4a.png)

**Comment by Laremere** [comment]
Using github latest fixed it, thanks!
